packages/convergence/src/convergence.js

```diff
@@ -0,0 +1,213 @@
+    Object.defineProperties(this, {
+      _timeout: { value: timeout },
+      _stack: { value: stack }
+    });
+  }
+
+  /**
+   * Creates a new convergence with the given timout; or if no timeout
+   * is given, will return the current timeout for this
+   * convergence. The new convergence's stack will be inherited
+   * from this convergence instance
+   *
+   * @param {Number} [timeout] - a timeout to create a new convergence with
+   * @returns {Number|Convergence} the current timeout or a new
+   * convergence instance
+   */
+  timeout(timeout) {
+    if (typeof timeout !== 'undefined') {
+      return new Convergence(timeout, [...this._stack]);
+    } else {
+      return this._timeout;
+    }
+  }
+
+  /**
+   * Creates a new convergence with the given assertion added to its
+   * stack. The new convergence's initial stack will be inherited from
+   * this convergence instance. The assertion given to `.and()` will
+   * be converged on using the `convergeOn` helper, but it's timeout
+   * will be managed by this convergence instance
+   *
+   * @param {Function} assert - the assertion to converge on
+   * @returns {Convergence} a new convergence instance
+   */
+  and(assert) {
+    return new Convergence(this._timeout, [
+      ...this._stack,
+      { assert }
+    ]);
+  }
+
+  /**
+   * Similar to `.and()`, creates a new convergence with the given
+   * assertion added to its stack. However, the assertion given to
+   * `.still()` will become inverted via the `convergeOn` helper.
+   * When a `.still()` is last in a stack, it will use the remaining
+   * convergence time to ensure the assertion continues to pass. The
+   * timeout is used when there are more assertions later in the
+   * stack, in which case this assertion needs to finish with enough
+   * time for other possible assertions to also run
+   *
+   * @param {Function} assert - the assertion to converge on
+   * @param {Number} [timeout] - this assertions timeout unless
+   * this assertion occurs at the end of the stack; defaults to 1/10
+   * of the total timeout (minimum 20ms)
+   * @returns {Convergence} a new convergence instance
+   */
+  still(assert, timeout) {
+    timeout = Math.max(timeout || (this._timeout / 10), 20);
+
+    return new Convergence(this._timeout, [
+      ...this._stack,
+      { assert, timeout, invert: true }
+    ]);
+  }
+
+  /**
+   * Creates a new convergence with the given callback added to its
+   * stack. The new convergence's initial stack will be inherited from
+   * this convergence instance. When a running convergence instance
+   * encounters a `.tap()`, it will invoke the callback with the value
+   * returned from the last function in the stack. The resulting
+   * return value will also be provided to the following function in
+   * the stack
+   *
+   * @param {Function} exec - the callback to execute during the convergence
+   * @returns {Convergence} a new convergence instance
+   */
+  tap(exec) {
```

Comment:
How do you see the primary use-case of `tap`. As a debugging tool?

Reply:
Debugging, and sometimes we might need to do something like this: 

```js
new Convergence()
  .and(() => expect(Page.$submit).to.exist)
  .tap(() => Page.submitForm(...))
  .still(() => expect(Page.$newItem).to.exist)
```

Then make assertions on this new item